goshopify.go

```diff
@@ -205,6 +210,12 @@ func NewClient(app App, shopName, token string) *Client {
 	return c
 }
 
+// WithLogger just sets logger and returns the Client.
+func (c *Client) WithLogger(l Logger) *Client {
```

Comment:
using l (lower-case L) as a name is not recommended. too easy to confuse  with I (upper-case eye) and 1 (one) depending on the font. Not a big deal here, but still...